Added complex sparse matrix instantiations.

MalikScheifinger opened the pull request:
Added instantiations to copy real matrices into complex matrices and performing matrix vector multiplications of a complex matrix with a real vector into a complex vector.

bangerth reviewed (commented):
Yes, very nice! Would you mind adding an entry to `doc/news/changes/minor` that documents the new ability?

bangerth commented:
@MalikScheifinger Ping?

MalikScheifinger commented:
I am sorry. Now it is done.

MalikScheifinger commented:
Well it seems like I never really tested it. I only ran the whole testsuite and nothing failed. But somehow the new test was not included. I fixed this and changed the tests output a little bit and included the output file now.

masterleinad commented:
```
Invalid author 'MScheifinger' without firstname and lastname

For possible solutions, consult the following page:
      https://github.com/dealii/dealii/wiki/Commit-authorship
```

MalikScheifinger commented:
I did a rebase and changed the author name in one commit.

masterleinad commented:
You still have to fix the indentation
```diff
diff --git a/tests/lac/complex_sparse_matrix.cc b/tests/lac/complex_sparse_matrix.cc
index f00974ec2..f2c1e7683 100644
--- a/tests/lac/complex_sparse_matrix.cc
+++ b/tests/lac/complex_sparse_matrix.cc
@@ -38,7 +38,8 @@ check_operations()
   SparseMatrix<std::complex<number>> cmpl_matrix(pattern);
 
   real_matrix.set(0, 0, 1.0);
-  deallog << "real_matrix.set(0, 0, 1.0) " << (real_matrix(0, 0) == 1.0 ? "True" : "False")  << std::endl;
+  deallog << "real_matrix.set(0, 0, 1.0) "
+          << (real_matrix(0, 0) == 1.0 ? "True" : "False") << std::endl;
 
 
   cmpl_matrix.copy_from(real_matrix);
```
